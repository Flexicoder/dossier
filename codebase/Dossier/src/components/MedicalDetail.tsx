import React, { Component } from 'react';
import { TextInput, View } from 'react-native';
import  styles  from '../assets/styles'

interface MedicalDetailProps {
    treatments?: string | null;
    consultant?: string | null;
    clinicalNurseSpecialist?: string | null;
    diagnosis?: string | null;
}

export default class MedicalDetail extends Component<MedicalDetailProps> {
    constructor(props: MedicalDetailProps) {
        super(props)
        this.state = { name: props.treatments,
                       relationship: props.consultant,
                       address: props.clinicalNurseSpecialist,
                       contactNumber: props.diagnosis }
    }

    render() {
        return (
        <View style={styles.container}>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Treatments"
                            multiline={true}
                            onChangeText={text => this.setState({ treatments: text})}
                            defaultValue={this.props.treatments || ""} />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Consultant"
                            onChangeText={text => this.setState( { consultant: text })}
                            defaultValue={this.props.consultant || ""}
                            />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Clinical Nurse Specialist"
                            onChangeText={text => this.setState( { clinicalNurseSpecialist: text })}
                            defaultValue={this.props.clinicalNurseSpecialist || ""}
                            />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Diagnosis"
                            multiline={true}
                            onChangeText={text => this.setState( { diagnosis: text })}
                            defaultValue={this.props.diagnosis || ""}
                            />
                    </View>
                </View>
            </View>
        );
    }
}