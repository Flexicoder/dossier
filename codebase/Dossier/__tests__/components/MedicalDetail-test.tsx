
import 'react-native';
import React from 'react';
import MedicalDetail from '../../src/components/MedicalDetail';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe("MedicalDetail", () => {
    it('renders correctly', () => {
        const medicalDetail = renderer.create(<MedicalDetail />);
        expect(medicalDetail.toJSON()).toMatchSnapshot();
      });
})


describe("MedicalDetail", () => {
    it('contains supplied data', () => {
        const medicalDetail = renderer.create(<MedicalDetail treatments="Chemotherapy" consultant="Mary" clinicalNurseSpecialist="Tom" diagnosis="Cancer" />);
        const stringVersion = JSON.stringify(medicalDetail.toJSON())
        expect(stringVersion).toMatch(/Chemotherapy/);
        expect(stringVersion).toMatch(/Mary/);
        expect(stringVersion).toMatch(/Tom/);
        expect(stringVersion).toMatch(/Cancer/);
    });
})