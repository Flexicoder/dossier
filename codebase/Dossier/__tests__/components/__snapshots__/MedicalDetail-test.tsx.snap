// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicalDetail renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        multiline={true}
        onChangeText={[Function]}
        placeholder="Treatments"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Consultant"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Clinical Nurse Specialist"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        multiline={true}
        onChangeText={[Function]}
        placeholder="Diagnosis"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
